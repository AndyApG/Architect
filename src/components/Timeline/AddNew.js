import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from '../../ui/src/components';
import { Zoom } from '../../behaviours';
import { getCSSVariableAsString } from '../../utils/CSSVariables';

const zoomColors = [getCSSVariableAsString('--light-background'), getCSSVariableAsString('--light-background')];

const AddNewButton = Zoom(
  ({ onInsertStage }) => (
    <button
      className="timeline-add-new__button"
      onClick={onInsertStage}
      style={{ display: 'inline-block' }}
    >
      <div className="timeline-add-new__button-label">Add new stage here</div>
      <Icon name="add-a-screen" className="timeline-add-new__button-icon" />
    </button>
  ),
);

const TimelineAddNew = ({ onInsertStage }) => (
  <div className="timeline-add-new">
    <AddNewButton
      zoomColors={zoomColors}
      onInsertStage={onInsertStage}
    />
  </div>
);

TimelineAddNew.propTypes = {
  onInsertStage: PropTypes.func.isRequired,
};

export default TimelineAddNew;
