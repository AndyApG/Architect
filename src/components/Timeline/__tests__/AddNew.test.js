/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import AddNew from '../AddNew';

jest.mock('../../../utils/CSSVariables');

const mockProps = {
  onInsertStage: () => {},
};

describe('<AddNew />', () => {
  it('can render', () => {
    const component = shallow(<AddNew {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
