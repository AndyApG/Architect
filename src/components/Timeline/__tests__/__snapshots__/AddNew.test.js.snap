// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddNew /> can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineAddNew
    onInsertStage={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <getContext(Draft(Component))
        onInsertStage={[Function]}
        zoomColors={
                Array [
                        "#227733",
                        "#227733",
                      ]
        }
/>,
      "className": "timeline-add-new",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "onInsertStage": [Function],
        "zoomColors": Array [
          "#227733",
          "#227733",
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <getContext(Draft(Component))
          onInsertStage={[Function]}
          zoomColors={
                    Array [
                              "#227733",
                              "#227733",
                            ]
          }
/>,
        "className": "timeline-add-new",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "onInsertStage": [Function],
          "zoomColors": Array [
            "#227733",
            "#227733",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
